### PR TITLE
WIP check gradients

### PR DESCRIPTION
**Pull request by jasmainak**
Here's a quick sketch of something I'd like to do. Basically it's a way for us to test the gradients ...

@pavanramkumar let me know if you are okay with changes along this line. Otherwise, it's something we can discuss during the sprint.

### CODE REVIEW

**Comment by pavanramkumar** [comment]
I like it. Only con is adding a dependency to `scipy`. I'm ok with that. We can implement it at the sprint.

**Comment by jasmainak** [comment]
Ha, I didn't realize we didn't have a dependency on `scipy` ;) I think `scipy` is a rather cheap dependency to have ...

**Comment by jasmainak** [comment]
The other advantage is that we can use `scipy` solvers easily if we have the gradient readily available and separate from the class ...

**Comment by jasmainak** [comment]
Wait, don't we already have a dependency of scipy thanks to this: https://github.com/glm-tools/pyglmnet/blob/master/pyglmnet/pyglmnet.py#L7 ?

File: pyglmnet/pyglmnet.py [hunk 1]
```diff
@@ -38,6 +38,175 @@ def set_log_level(verbose):
...
+    elif distr == 'multinomial':
+        # this assumes that y is already as a one-hot encoding
+        pred = _qu(distr, z, eta)
+        grad_beta0 = -1. / n_samples * np.sum(y - pred, axis=0)
+        grad_beta = -1. / n_samples * \
+            np.transpose(np.dot(np.transpose(y - pred), X)) \
+            + reg_lambda * (1 - alpha) * \
+            np.dot(InvCov, beta[1:])
+
+    n_classes = y.shape[1] if distr == 'multinomial' else 1
```
Comment (jasmainak) [inline pyglmnet/pyglmnet.py]: @pavanramkumar is the grad for `multinomial` case also going to be 2D? I think you can vectorize it and then reshape it outside of the loss function

**Comment by jasmainak** [comment]
@pavanramkumar for now I moved the `test_gradients` to the top so that we make sure it passes first. Can you check why the test fails? Right now I get a difference of 0.5 between the real gradient and the computed gradient. It's too large ...

**Pull request merged**